stubs/toml/toml/decoder.pyi

```diff
@@ -0,0 +1,78 @@
+import sys
+from _typeshed import StrPath
+from typing import IO, Callable, Generic, MutableMapping, Pattern, Protocol, Text, TypeVar
+
+_T = TypeVar("_T")
+_KT_contra = TypeVar("_KT_contra", contravariant=True)
+_VT = TypeVar("_VT")
+
+if sys.version_info >= (3, 3):
+    FNFError = FileNotFoundError
+else:
+    FNFError = IOError
+
+if sys.version_info >= (3, 6):
+    _PathLike = StrPath
+elif sys.version_info >= (3, 4):
+    import pathlib
+
+    _PathLike = StrPath | pathlib.PurePath
+else:
+    _PathLike = StrPath
+
+class _SupportsGetSetItem(Protocol[_KT_contra, _VT]):
+    def __getitem__(self, __k: _KT_contra) -> _VT: ...
+    def __setitem__(self, __k: _KT_contra, __v: _VT) -> None: ...
+
+TIME_RE: Pattern[str]
+
+class TomlDecodeError(ValueError):
+    msg: str
+    doc: str
+    pos: int
+    lineno: int
+    colno: int
+    def __init__(self, msg: str, doc: str, pos: int) -> None: ...
+
+class CommentValue(_SupportsGetSetItem[_KT_contra, _VT]):
+    val: _SupportsGetSetItem[_KT_contra, _VT]
+    comment: str
+    def __init__(
+        self,
+        val: _SupportsGetSetItem[_KT_contra, _VT],
+        comment: str,
+        beginline: bool,
+        _dict: type[MutableMapping[_KT_contra, _VT]],
+    ) -> None: ...
+    def __getitem__(self, key: _KT_contra) -> _VT: ...
+    def __setitem__(self, key: _KT_contra, value: _VT) -> None: ...
+    def dump(self, dump_value_func: Callable[[_SupportsGetSetItem[_KT_contra, _VT]], str]) -> str: ...
```

Comment:
This will make defining a callback function quite difficult. You can't annotate it as taking a dict, even if it will actually get passed a dict, because this annotation says that it must be compatible with any `_SupportsGetSetItem` object.

Maybe this class should be generic over a mapping type, rather than separate key and value types?

Reply:
What do you think about the last commit? :)